Source/WebCore/animation/KeyframeEffect.cpp

```diff
@@ -2238,6 +2239,36 @@ void KeyframeEffect::wasAddedToEffectStack()
 void KeyframeEffect::wasRemovedFromEffectStack()
 {
     m_inTargetEffectStack = false;
+
+    if (!canBeAccelerated())
+        return;
+
+#if ENABLE(THREADED_ANIMATIONS)
+    if (canHaveAcceleratedRepresentation())
+        return;
+#endif
+
+    // If the effect was running accelerated, we need to mark it for removal straight away
+    // since it will not be invalidated by a future call to KeyframeEffectStack::applyPendingAcceleratedActions().
+    if (animation() && (isRunningAccelerated() || isAboutToRunAccelerated())) {
+        auto* anim = animation();
```

Comment:
I'm surprised the safer-cpp bot didn't complain about this not being ref-counted since `WebAnimation::playState()` is not a trivial function. We typically don't shorten names like `anim` instead of `animation`.
```suggestion
        Ref animation = *animation();
```